enhancements/cluster-logging/forwarder-input-selectors.md

```diff
@@ -0,0 +1,285 @@
+---
+title: forwarder-input-selectors
+authors:
+- "@jcantril"
+reviewers:
+- "@alanconway, Red Hat Logging Architect"
+- "@xperimental"
+- "@syedriko"
+ - "@cahartma"
+approvers:
+- "@alanconway"
+api-approvers:
+- "@alanconway"
+creation-date: 2023-10-30
+last-updated: 2023-10-30
+tracking-link:
+- https://issues.redhat.com/browse/LOG-2155
+see-also:
+-
+replaces:
+-
+---
+
+
+# Log Forwarding Input Slection using Kubernetes Metadata
+## Summary
+
+
+Cluster Logging defines a set of well known log sources in order to facilitate configuration of log collection and normalization.  Given customers are no longer bound to the data storage provided by cluster logging, this enhancement expands those definitions to allow specifying which logs are collected by using Kubernetes metadata.
+
+
+Logs originate from six distinct sources and are logically grouped using the following definitions:
+
+
+* **Application** are container logs from all namespaces across the cluster excluding infrastructure namespaces.  
+
+
+* **Infrastructure** logs are:
+  * container logs from namespaces: default, kube*, openshift*
+
+
+* **Audit** are logs written to files on master nodes that include:
+  * kubernetes API server
+  * OpenShift API server
+  * auditd
+  * OVN
+
+
+**NOTE**: **application**, **infrastructure**, and **audit** are reserved words to the **cluster-logging-operator** and continue to represent the previous definitions.
+
+
+Administrators use these definitions to specify pipelines to normalize and route messages from the sources to outputs.
+
+
+This enhancement allow administrators to define "named" inputs by expanding the previous definitions as follows:
+
+
+* Named application:
+  * Any name that is not reserved
+  * Collect from any namespace including the ones for **infrastructure** container logs
+* Named infrastructure:
+  * Any name that is not reserved
+  * Explicit source choices of: node, container
+* Named audit:
+  * Any name that is not reserved
+  * Explicit source choices of: kubeAPI, openshiftAPI, auditd, ovn
+
+
+
+
+## Motivation
+
+
+### User Stories
+
+
+
+
+* As an administrator of cluster logging, I want to only forward logs from a limited set of namespaces because I do not need the others
+* As an administrator of cluster logging, I want to exclude logs from a limited set of namespaces because I do not need them
+* As an administrator of cluster logging, I want to only forward logs from pods with a specific set of labels
+* As an administrator of cluster logging, I want to exclude certain container logs from a pod because they are noisy and uninteresting to me
+* As an administrator of cluster logging, I do not want to collect node logs because they are not of interest to me
+
+
+### Goals
```

Comment:
I'd add these other two points:

*  Reduce the network usage when log forwarding
*  Reduce the resources required on the log store side related to storage size, cpu and memory